Clarify when (#3697)
modelica/ModelicaSpecification: chapters/classes.tex

A \lstinline!Real! variable assigned in a \lstinline!when!-clause is a discrete-time variable, even though it was not declared with the prefix \lstinline!discrete!.
A \lstinline!Real! variable not assigned in any \lstinline!when!-clause and without any type prefix is a continuous-time variable.

The determination of whether a variable is assigned in \lstinline!when!-clause requires evaluating the controlling conditions of any enclosing \lstinline!if!-equations and \lstinline!for!-equations, \cref{where-a-when-equation-may-occur}.
\begin{nonnormative}

> Checking whether a \lstinline!Real! variable declared as \lstinline!discrete! is assigned in a when-clause for other valid values of controlling conditions is a quality of implementation issue for tools.

Reviewer: Formatting:
```suggestion
Checking whether a \lstinline!Real! variable declared as \lstinline!discrete! is assigned in a \lstinline!when!-clause for other valid values of controlling conditions is a quality of implementation issue for tools.
```

> Checking whether a \lstinline!Real! variable declared as \lstinline!discrete! is assigned in a \lstinline!when!-clause for other valid values of controlling conditions is a quality of implementation issue for tools.

Reviewer: I don't think we every say _… for tools_ when talking out quality of implementation:
```suggestion
It is a quality of implementation to check that a variable declared as \lstinline!discrete! is assigned in a \lstinline!when!-clause also for other valid values of controlling conditions.
```
